modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/spaces/SpaceSummaryHeader.tsx

```diff
@@ -39,14 +44,63 @@ interface SpaceSummaryHeaderProps {
 }
 
 export default function SpaceSummaryHeader({
+	creationMenu,
 	label,
 	permissions,
 	spaceModalProps,
 	title,
 	url,
 }: SpaceSummaryHeaderProps) {
+	const [active, setActive] = useState(false);
+
 	const loadData = () => window.location.reload();
 
+	const CreationMenu = () => {
+		if (!creationMenu?.primaryItems?.length) {
+			return null;
+		}
+
+		return (
+			<ClayDropDown
+				active={active}
+				className="ml-3"
+				onActiveChange={setActive}
+				trigger={
+					<ClayButtonWithIcon
+						aria-label={`Add ${title}`}
+						displayType="secondary"
+						small
+						symbol="plus"
+						title={`Add ${title}`}
+					/>
+				}
+			>
+				<ClayDropDown.ItemList>
+					{addOnClickToCreationMenuItems(
+						creationMenu.primaryItems,
+						ACTIONS
+					).map((item: any, i: number) => (
```

Comment:
instead of `i` let's rename to `index`. 
And we should try to avoid typing with `any`, but I am not sure which type is it, could be `ItemData`?